Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1096,6 +1097,18 @@ public void run() {
 	}
 
 	/**
+	 * Indicates whether or not this user has a custom identity specified for them. Note that this is independent of installs.
+	 * If you call setIdentity, this device will have that identity associated with this user until logout is called.
+	 * This includes persisting through uninstalls, as we track device id.
+	 *
+	 * @return 				A {@link Boolean} value that will return <i>true</i> only if user already has an identity.
+	 *
+	 */
+	public boolean isUserIdentified() {
+		return !prefHelper_.getIdentity().equals(PrefHelper.NO_STRING_VALUE);
```

Comment:
hmm,For preference  I guess it may be a good idea   to have a default value like this.It will help to differentiate between the condition
1) "Key" is actually not present in the preference
2)Value for the key is intentionally set to NULL.
  Any way the API for getting a String from preference actually need a default value (can be specified as null too)